nautilus_trader/risk/greeks.pyx

```diff
@@ -0,0 +1,260 @@
+# -------------------------------------------------------------------------------------------------
+#  Copyright (C) 2015-2025 Nautech Systems Pty Ltd. All rights reserved.
+#  https://nautechsystems.io
+#
+#  Licensed under the GNU Lesser General Public License Version 3.0 (the "License");
+#  You may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at https://www.gnu.org/licenses/lgpl-3.0.en.html
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+# -------------------------------------------------------------------------------------------------
+
+from typing import Callable
+
+from nautilus_trader.core.nautilus_pyo3 import black_scholes_greeks
+from nautilus_trader.core.nautilus_pyo3 import imply_vol_and_greeks
+from nautilus_trader.model.enums import InstrumentClass
+from nautilus_trader.model.enums import PriceType
+from nautilus_trader.model.greeks import GreeksData
+from nautilus_trader.model.greeks import InterestRateCurveData
+from nautilus_trader.model.greeks import PortfolioGreeks
+
+from nautilus_trader.cache.base cimport CacheFacade
+from nautilus_trader.common.component cimport Clock
+from nautilus_trader.common.component cimport Logger
+from nautilus_trader.common.component cimport MessageBus
+from nautilus_trader.core.datetime cimport unix_nanos_to_dt
+from nautilus_trader.core.rust.model cimport OptionKind
+from nautilus_trader.core.rust.model cimport PositionSide
+from nautilus_trader.model.data cimport DataType
+from nautilus_trader.model.identifiers cimport InstrumentId
+from nautilus_trader.model.identifiers cimport StrategyId
+from nautilus_trader.model.identifiers cimport Venue
+from nautilus_trader.model.objects cimport Price
+from nautilus_trader.model.position cimport Position
+
+
+cdef class GreeksCalculator:
+    def __init__(self, msgbus: MessageBus, cache: CacheFacade, clock: Clock, logger: Logger):
```

Comment:
Ideally we follow the formatting convention of other Cython classes (hang params).

Also, need to add docstrings.